config/peerpods/podvm/azure-podvm-image-handler.sh

```diff
@@ -718,6 +737,52 @@ function delete_image_using_id() {
     echo "Azure image deleted successfully"
 }
 
+# Function to check if image already exists
+# This checks if IMAGE_VERSION exist in Azure
+# and the LATEST_IMAGE_ID annotation is set in the peer-pods-cm configmap
+# 0: Image exists and matches
+# 1: Image does not exist
+# 2: Image mismatch (caller should delete Image)
+# The required variables are assumed to be set
+function image_exists() {
```

Comment:
I missed this method under gcp. IIUC, most of it could be generic to all providers?

Reply:
A portion of it yes, but querying is specific to individual provider. So I kept it for each provider..

Reply:
And where is the gcp implementation? I saw you changed the gcp-podvm-image-handler.sh but I could not find any image_exists method.